<?xml version="1.0" encoding="UTF-8"?>

<launch>

  <!-- ROS-Pybullet interface node simulator -->
  <node pkg="ros_pybullet_interface" name="ros_pybullet_interface_node" type="ros_pybullet_interface_node.py" output="screen">

    <!-- Config file for Pybullet -->
    <param name="pybullet_config" value="{rpbi_work}/configs/pybullet_nextage_pushing.yaml" type="str"/>

    <!-- Node frequency -->
    <param name="ros_node_freq" value="100" type="int"/>

    <!-- Simulated robot -->
    <rosparam param="robot_config_file_names">
      [
      ]
    </rosparam>

    <!-- Visual robot -->
    <rosparam param="visual_robot_config_file_names">
      [
      "{rpbi_work}/configs/nextage.yaml",
      ]
    </rosparam>

    <remap from="/rpbi/nextage/joint_state/target" to="/nextagea/joint_states"/>

    <rosparam param="dynamic_object_config_file_names">
      [
      "{rpbi_work}/configs/pushing_box.yaml",
      ]
    </rosparam>

    <rosparam param="collision_object_config_file_names">
      [
      "{rpbi_work}/configs/floor.yaml",
      "{rpbi_work}/configs/table.yaml",
      ]
    </rosparam>

    <rosparam param="visual_object_config_file_names">
      [
      "{rpbi_work}/configs/visual_goal_left.yaml",
      "{rpbi_work}/configs/visual_goal_right.yaml",
      ]
    </rosparam>

  </node>

  <!-- ROS-Pybullet interface node simulator -->
  <!-- <node pkg="ros_pybullet_interface" name="ros_pybullet_interface_node_second" type="ros_pybullet_interface_node.py" output="screen"> -->

  <!--   <!-\- Config file for Pybullet -\-> -->
  <!--   <param name="pybullet_config" value="{rpbi_work}/configs/pybullet_nextage_pushing.yaml" type="str"/> -->

  <!--   <!-\- Make listener node -\-> -->
  <!--   <param name="listener" value="true" type="bool"/> -->

  <!--   <!-\- Node frequency -\-> -->
  <!--   <param name="ros_node_freq" value="100" type="int"/> -->

  <!--   <!-\- Simulated robot -\-> -->
  <!--   <rosparam param="robot_config_file_names"> -->
  <!--     [ -->
  <!--     "{rpbi_work}/configs/nextage.yaml", -->
  <!--     ] -->
  <!--   </rosparam> -->

  <!--   <!-\- Visual robot -\-> -->
  <!--   <rosparam param="visual_robot_config_file_names"> -->
  <!--     [ -->
  <!--     ] -->
  <!--   </rosparam> -->

  <!--   <rosparam param="dynamic_object_config_file_names"> -->
  <!--     [ -->
  <!--     "{rpbi_work}/configs/pushing_box.yaml", -->
  <!--     ] -->
  <!--   </rosparam> -->

  <!--   <rosparam param="collision_object_config_file_names"> -->
  <!--     [ -->
  <!--     "{rpbi_work}/configs/floor.yaml", -->
  <!--     "{rpbi_work}/configs/table.yaml", -->
  <!--     ] -->
  <!--   </rosparam> -->

  <!--   <rosparam param="visual_object_config_file_names"> -->
  <!--     [ -->
  <!--     "{rpbi_work}/configs/visual_goal_left.yaml", -->
  <!--     "{rpbi_work}/configs/visual_goal_right.yaml", -->
  <!--     ] -->
  <!--   </rosparam> -->

  <!-- </node> -->

  <!-- Nextage pushing node (teleop) -->
  <node pkg="rpbi_work" name="nextage_pushing_node" type="nextage_pushing_node.py" output="screen"/>

  <!-- Static tf frames -->
  <node pkg="tf2_ros" type="static_transform_publisher" name="simulated_robot_base_tf_broadcaster"
	args="0. 0. 0.85 0 0 0 rpbi/world nextage_base"/>

  <node pkg="tf2_ros" type="static_transform_publisher" name="floor_tf_broadcaster"
	args="0 0 0 0 0 0 rpbi/world floor"/>

  <node pkg="tf2_ros" type="static_transform_publisher" name="table_top_tf_broadcaster"
	args="0.825 0 0.8 0 0 0 rpbi/world table_top"/>

  <node pkg="tf2_ros" type="static_transform_publisher" name="table_tf_broadcaster"
	args="0 0 -0.1 0 0 0 table_top table"/> <!-- DO NOT EDIT THIS -->

</launch>
